MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.cpp

```diff
@@ -35,9 +37,27 @@ PiecewiseLinearInterpolation::PiecewiseLinearInterpolation(
     }
 }
 
-double PiecewiseLinearInterpolation::getValue(double pnt_to_interpolate) const
+double PiecewiseLinearInterpolation::interpolate(
+    std::vector<double> const& supp_pnts,
+    std::vector<double> const& values_at_supp_pnts,
+    std::size_t const interval_idx,
+    double const pnt_to_interpolate) const
+{
+    assert(std::fabs(supp_pnts[interval_idx + 1] - supp_pnts[interval_idx]) >
```

Comment:
this depends on users input. i think this should be checked once in the constructor with OGS_FATAL.